[cmake] Incorrect gtest lib directory detection

When linking roc-cl-unittest, ld raises the error
```
ld: cannot find -lgtest
ld: cannot find -lgtest_main
```
despite the fact that the googletest library has just compiled and should be linked against. This seems to be due to an error in `src/unittest/CMakeLists.txt`. If I change line 66 from
```
link_directories(${binary_dir}/googletest)
```
to
```
link_directories(${binary_dir}/lib)
```
then it seems to link fine.

--- a/src/unittest/CMakeLists.txt
+++ b/src/unittest/CMakeLists.txt
@@ -63,7 +63,7 @@ ExternalProject_Add(
 ExternalProject_Get_Property(googletest source_dir)
 include_directories(${source_dir}/googletest/include)
 ExternalProject_Get_Property(googletest binary_dir)
-link_directories(${binary_dir}/googletest)
+link_directories(${binary_dir}/lib)
 
 #include(${GTEST_INCLUDE_DIRS})
 file(GLOB sources ${CMAKE_CURRENT_SOURCE_DIR}/*.cpp)

